nshlib/nsh_parse.c

```diff
@@ -1060,9 +1080,9 @@ static FAR char *nsh_cmdparm(FAR struct nsh_vtbl_s *vtbl, FAR char *cmdline,
 
   /* Create a unique file name using the task ID */
 
-  tmpfile = NULL;
-  ret = asprintf(&tmpfile, "%s/TMP%d.dat", CONFIG_LIBC_TMPDIR, getpid());
-  if (ret < 0 || !tmpfile)
+  ret = asprintf(&param_exec.file_out, "%s/TMP%d.dat",
```

Comment:
compile fail at 1126,
should free param_exec.file_out

Reply:
> compile fail at 1126, should free param_exec.file_out

`pipe2()` undefined, have add config `NSH_PIPELINE` that depends on `PIPES`.